### ISSUE TITLE
CreateWithInvitationScreen is calling callback twice when pressing top left arrow back button

### ISSUE THREAD

**Issue by kim-cv** [playgameservices/play-games-plugin-for-unity#1068]
PlayGamesPlatform.Instance.TurnBased.CreateWithInvitationScreen is calling callback twice when pressing top left arrow back button.

Google games plugin version: 0.9.31.a
Unity3D version: 5.3.3f1
Build minimum API: Android 4.0 Ice Cream Sandwich
Testing Device Android Version: 4.3

**Comment by claywilkinson** [comment]
Thanks.  If you can attach a log that will help fix the issue.

**Comment by kim-cv** [comment]
I'm unsure which type of log you refer to.

All I did was outputting to the unity console how many times the callback was called.
I also made sure by checking all references to all methods in my code to make sure the issue wasn't in my code :)

But if there is anything I can do to assist with this issue, please tell me how, I'll gladly help.

**Comment by kim-cv** [comment]
@claywilkinson I took the TickTackToe sample and I experience the same issue, I have also attached part of the log.

First line "Do Invite" is a `Debug.Log("Do invite");` right before PlayGamesPlatform.Instance.TurnBased.CreateWithInvitationScreen

It seems the internal callback is called twice at 04-04 16:04:17.819 and at 04-04 16:04:17.849

In the method "OnMatchStarted" I call `Debug.Log("OnMatchStarted succes: " + success);` as the first thing and you can see it being called in the log at 04-04 16:04:18.039 and at 04-04 16:04:18.049.

```
04-04 16:04:14.089: D/Unity(5333): Do invite: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:14.099: I/Unity(5333): UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
04-04 16:04:14.099: I/Unity(5333): UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
04-04 16:04:14.099: I/Unity(5333): UnityEngine.Logger:Log(LogType, Object)
04-04 16:04:14.099: I/Unity(5333): UnityEngine.Debug:Log(Object)
04-04 16:04:14.099: I/Unity(5333): MainMenuGui:DoGUI() (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\TicTacToe\Gui\MainMenuGui.cs:141)
04-04 16:04:14.099: I/Unity(5333): BaseGui:OnGUI() (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\TicTacToe\Gui\BaseGui.cs:94)
04-04 16:04:14.109: D/Unity(5333):  [Play Games Plugin DLL] 04/04/16 16:04:14 +02:00 DEBUG: Entering internal callback for AndroidPlatformConfiguration#InternalIntentHandler: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:14.109: I/Unity(5333): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 37)
04-04 16:04:14.129: D/NativeBridgeActivity(5333): Launching bridge activity: parent:com.unity3d.player.UnityPlayerActivity@21a870f0 intent Intent { act=com.google.android.gms.games.SELECT_OPPONENTS_TURN_BASED dat=version:8487000 pkg=com.google.android.play.games (has extras) }
04-04 16:04:14.179: D/Unity(5333):  [Play Games Plugin DLL] 04/04/16 16:04:14 +02:00 DEBUG: Application is pausing, which disconnects the RTMP  client.  Leaving room.: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:14.179: I/Unity(5333): UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
04-04 16:04:14.179: I/Unity(5333): UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
04-04 16:04:14.179: I/Unity(5333): UnityEngine.Logger:Log(LogType, Object)
04-04 16:04:14.179: I/Unity(5333): UnityEngine.Debug:Log(Object)
04-04 16:04:14.179: I/Unity(5333): GooglePlayGames.OurUtils.Logger:d(String) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\OurUtils\Logger.cs:59)
04-04 16:04:14.179: I/Unity(5333): GooglePlayGames.Native.NativeRealtimeMultiplayerClient:HandleAppPausing(Boolean) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\Platforms\Native\NativeRealtimeMultiplayerClient.cs:123)
04-04 16:04:14.179: I/Unity(5333): GooglePlayGames.OurUtils.PlayGamesHelperObject:OnApplicationPause(Boolean) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\OurUtils\PlayGamesHelperObject.cs:162)
04-04 16:04:17.819: D/Unity(5333):  [Play Games Plugin DLL] 04/04/16 16:04:17 +02:00 DEBUG: Entering internal callback for TurnBasedManager#PlayerSelectUICallback: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:17.819: I/Unity(5333): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 37)
04-04 16:04:17.849: D/Unity(5333):  [Play Games Plugin DLL] 04/04/16 16:04:17 +02:00 DEBUG: Entering internal callback for TurnBasedManager#InternalTurnBasedMatchCallback: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:17.849: I/Unity(5333): (Filename: ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp Line: 37)
04-04 16:04:18.039: D/Unity(5333): OnMatchStarted succes: False: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:18.049: I/Unity(5333): UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
04-04 16:04:18.049: I/Unity(5333): UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
04-04 16:04:18.049: I/Unity(5333): UnityEngine.Logger:Log(LogType, Object)
04-04 16:04:18.049: I/Unity(5333): UnityEngine.Debug:Log(Object)
04-04 16:04:18.049: I/Unity(5333): MainMenuGui:OnMatchStarted(Boolean, TurnBasedMatch) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\TicTacToe\Gui\MainMenuGui.cs:60)
04-04 16:04:18.049: I/Unity(5333): GooglePlayGames.Native.<CreateWithInvitationScreen>c__AnonStorey55:<>m__78(UIStatus, TurnBasedMatch) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\Platforms\Native\NativeTurnBasedMultiplayerClient.cs:94)
04-04 16:04:18.049: I/Unity(5333): GooglePlayGames.Native.PInvoke.<AsOnGameThreadCallback>c__AnonStorey6D`2:<>m__A1() (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\Platforms\Native\PInvoke\Callbacks.cs:231)
04-04 16:04:18.049: I/Unity(5333): GooglePlayGames.OurUtils.PlayGamesHelperObject:Update() (at C:\GoogleTickT
04-04 16:04:18.049: D/Unity(5333): OnMatchStarted succes: False: 0 at ./artifacts/generated/common/runtime/UnityEngineDebugBindings.gen.cpp:37 (17408, 0, 0)
04-04 16:04:18.049: I/Unity(5333): UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
04-04 16:04:18.049: I/Unity(5333): UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
04-04 16:04:18.049: I/Unity(5333): UnityEngine.Logger:Log(LogType, Object)
04-04 16:04:18.049: I/Unity(5333): UnityEngine.Debug:Log(Object)
04-04 16:04:18.049: I/Unity(5333): MainMenuGui:OnMatchStarted(Boolean, TurnBasedMatch) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\TicTacToe\Gui\MainMenuGui.cs:60)
04-04 16:04:18.049: I/Unity(5333): GooglePlayGames.Native.<CreateWithInvitationScreen>c__AnonStorey55:<>m__78(UIStatus, TurnBasedMatch) (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\Platforms\Native\NativeTurnBasedMultiplayerClient.cs:94)
04-04 16:04:18.049: I/Unity(5333): GooglePlayGames.Native.PInvoke.<AsOnGameThreadCallback>c__AnonStorey6D`2:<>m__A1() (at C:\GoogleTickTackToe\GoogleTickTackToe\Assets\GooglePlayGames\Platforms\Native\PInvoke\Callbacks.cs:231)
04-04 16:04:18.049: I/Unity(5333): GooglePlayGames.OurUtils.PlayGamesHelperObject:Update() (at C:\GoogleTickT
```

**Comment by claywilkinson** [comment]
Thanks for the log.

**Comment by claywilkinson** [comment]
I'll work on updating the plugin - but in the mean time you can add one line to fix it:
In  Assets/GooglePlayGames/Platforms/Native/NativeTurnBasedMultiplayerClient.cs, line 105, right after calling the callback, add `return;`

**Comment by claywilkinson** [comment]
Fix merged into 0.9.33
